src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs

```diff
@@ -0,0 +1,568 @@
+            yield return new object[] { "Content-Type:charset=unicode", Encoding.Default };
+
+            yield return new object[] { "Content-Type:", Encoding.Default };
+            yield return new object[] { "Content-Type:;", Encoding.Default };
+            yield return new object[] { "Content-Type:application/json", Encoding.Default };
+            yield return new object[] { "Content-Type:application/json;", Encoding.Default };
+            yield return new object[] { "Content-Type:application.json;charset", Encoding.Default };
+            yield return new object[] { "Content-Type:application.json;charset  ", Encoding.Default };
+            yield return new object[] { "Content-Type:application.json;charset;", Encoding.Default };
+            yield return new object[] { "Content-Type:application.json;charset=", Encoding.Default };
+            yield return new object[] { "Content-Type:application.json;charset=\"", Encoding.Default };
+            yield return new object[] { "Content-Type:application.json;charset=,", Encoding.Default };
+            yield return new object[] { "Content-Type:application.json;charset=\"unicode", Encoding.Default };
+            yield return new object[] { "Content-Type:application/json;charset=NoSuchEncoding", Encoding.Default };
+            yield return new object[] { "Content-Type:application/json;charset=unicode; boundary=something", Encoding.Default };
+
+            yield return new object[] { "Unknown-Header: Test", Encoding.Default };
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        [MemberData(nameof(ContentEncoding_TestData))]
+        public async Task ContentEncoding_Get_ReturnsExpected(string header, Encoding expected)
+        {
+            if (!PlatformDetection.IsWindows)
+            {
+                expected = Encoding.Default; // ContentEncoding is never set.
+            }
+
+            await GetRequest("POST", "", new string[] { header }, (_, request) =>
+            {
+                Assert.Equal(expected, request.ContentEncoding);
+            });
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        public async Task ContentEncoding_NoBody_ReturnsDefault()
+        {
+            await GetRequest("POST", "", new string[] { "Content-Length: 0", "Content-Type:application/json;charset=unicode" }, (_, request) =>
+            {
+                Assert.Equal(Encoding.Default, request.ContentEncoding);
+            }, sendContent: false);
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        [InlineData("Content-Length: 0", 0)]
+        [InlineData("Transfer-Encoding: chunked", -1)]
+        public async Task ContentLength_Get_ReturnsExpected(string contentLengthString, long expected)
```

Comment:
In all of these test names, what is the "Get" referring to?  Curious because it's using a "POST" method rather than "GET".  Is it meant to refer to getting the value from the property?